### ISSUE TITLE
Extracting public transport routes with associated ordered stops

### ISSUE THREAD

**Issue by mikhailsirenko** [HTenkanen/pyrosm#43]
Hi Henrikki,

I managed to extract tram lines and tram stops as was shown in the docs, but I cannot understand how to get the relations between a single line and its set of stops and their order. E.g. tram line 1 has 10 tram stops, the first stop is 1, the second 2, etc. Similarly to this https://www.openstreetmap.org/relation/721888. 

Perhaps, there should be a different query for that?

Thanks in advance!

**Comment by HTenkanen** [comment]
Hi @mikhailsirenko, and thanks for this question! 👍 It's an interesting one.. My first thoughts are that it might require a bit of extra tricks to get the stop numbers in correct order, because the nodes that represent the stops typically does not include any tags that would inform about their sequence number, such as this which is part of the route that you gave as an example: https://www.openstreetmap.org/node/706566618

Hence, you might need to do some GIS tricks to assign and keep track of each stops position along the tram line, and then count from either side of the line that what is the sequence number of a given stop.. That said, it might be much easier just to use GTFS data e.g. from https://transitfeeds.com/ that contains the sequence number of the stops as attribute information. Have you considered this option?

It would be very interesting to also be able to solve this ordering problem using purely OSM data, but as Pyrosm aims to be quite generic OSM parsing library, my initial thoughts are that it might be better to write a separate tool for this purpose and not to directly integrate it to Pyrosm. My feeling is that doing this kind of map matching task might get quite complex requiring a lot of customization.

**Comment by HTenkanen** [comment]
It seems that Transitfeeds might not have the data for Dresden specifically, but I think this national level source might also have the data that you need? --> https://gtfs.de/en/feeds/de_nv/

**Comment by mikhailsirenko** [comment]
Thanks for the reply @HTenkanen!

Yup, I had the same thoughts. I've tried to count the stops using directions, but the problems are:
- Some of the stops don't have a route reference (e.g. route ref key is absent in the tags dict);
- Route directions (start and end) aren't equal to any of the stop names;
- Routes can bi-directional and stops aren't associated with any.

And indeed I found the "solution" for my "problem" in GTFS data. Would be cool, of course, to if it can be solved with OSM only, but seems it is not. Maybe there is a need for such a combination: GTFS and OSM?
